src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -245,9 +270,23 @@ export class PostgresQueryRunner
         }
         this.transactionDepth -= 1
 
+        await Promise.all(this.afterRollbackListeners.map((listener) => listener())).finally(() => {
+            this.afterRollbackListeners = [];
+            this.afterCommitListeners = [];
+        });
+        
         await this.broadcaster.broadcast("AfterTransactionRollback")
     }
 
+    /**
+     * Rollbacks transaction if transaction was not started do nothing
+     */
+    async rollbackTransactionIfNotStarted(): Promise<void> {
+        if (!this.isTransactionActive) return
+        
+        return this.rollbackTransaction();
+    }
```

Comment:
This is basically only muting the error that `rollbackTransaction` would throw the this.isTransactionActive is false.
Is this what you expected?

Reply:
yes

Reply:
What don't you simply trycatch `rollbackTransaction` instead?
Can you give an example when you want to do a rollback outside of a transaction?